app/src/features/Batches/BatchCreate.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from "react";
+
+import {
+  Button,
+  Chip,
+  FormControl,
+  Input,
+  InputLabel,
+  MenuItem,
+  Select,
+  Typography,
+  useTheme,
+} from "@material-ui/core";
+import { makeStyles, Theme } from "@material-ui/core/styles";
+import PlayIcon from "@material-ui/icons/PlayCircleOutline";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router-dom";
+
+import {
+  useApiResourcesListQuery,
+  useApiBatchesCreateMutation,
+} from "services/api/endpoints";
+import type { Resource } from "services/api/generated/api.generated";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    flexDirection: "row",
+  },
+  formControl: {
+    margin: theme.spacing(1),
+    minWidth: 120,
+    maxWidth: 300,
+  },
+  button: {
+    margin: theme.spacing(1),
+    marginTop: "auto",
+  },
+  chips: {
+    display: "flex",
+    flexWrap: "wrap",
+  },
+  chip: {
+    margin: 2,
+  },
+}));
+
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+  PaperProps: {
+    style: {
+      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+      width: 250,
+    },
+  },
+};
+
+const getStyles = (resourceId: string, resourceIds: string[], theme: Theme) => {
+  return {
+    fontWeight: resourceIds.includes(resourceId)
+      ? theme.typography.fontWeightRegular
+      : theme.typography.fontWeightMedium,
+  };
+};
+
+const BatchCreate = (): JSX.Element => {
+  const classes = useStyles();
+  const theme = useTheme();
+  const { t } = useTranslation();
+
+  const [selectedResourceIds, setSelectedResourceIds] = useState<string[]>([]);
+
+  const { sourceId: id } = useParams<{ sourceId: string }>();
+
+  const { data: resources } = useApiResourcesListQuery(
+    { source: id },
+    { skip: !Boolean(id) }
+  );
+
+  const [apiBatchCreate] = useApiBatchesCreateMutation();
+
+  const handleResourceSelectionChange = (
+    event: React.ChangeEvent<{
+      name?: string | undefined;
+      value: unknown;
+    }>
+  ) => {
+    setSelectedResourceIds(event.target.value as string[]);
+  };
+
+  const handleBatchRun = () => {
+    if (!selectedResourceIds.length) return;
+
+    apiBatchCreate({
+      batchRequest: {
+        resources: selectedResourceIds.map((id) => ({
+          resource_id: id,
+        })),
+      },
+    });
```

Comment:
is the error handled somehow if this fails ? it doesn't seem to be the case but maybe I'm missing something on error handling

Reply:
I'm letting this on the side for the moment, the API could change. But yes I need to handle the errors.